src/drf_pydantic/fields.py

```diff
@@ -0,0 +1,52 @@
+from enum import Enum
+from typing import Type, Optional
+
+from rest_framework.fields import empty
+from rest_framework.serializers import ChoiceField
+
+
+class EnumField(ChoiceField):
+    """
+    Custom DRF field that restricts accepted values to that of a defined enum
+    """
+    default_error_messages = {'invalid': 'No matching enum type'}
+
+    def __init__(self, enum: Type[Enum], **kwargs):
+        self.enum = enum
+        kwargs.setdefault('choices', [(x, x.name) for x in self.enum])
+        super().__init__(**kwargs)
+
+    def run_validation(self, data=empty) -> Optional[Enum]:
```

Comment:
Can you add a type hint for `data` argument

Reply:
Added